lib/DxcSupport/FileIOHelper.cpp

```diff
@@ -668,34 +678,50 @@ HRESULT DxcCreateBlob(
     pData = heapCopy.m_pData;
     if (pData == nullptr)
       return E_OUTOFMEMORY;
-    if (pPtr)
-      memcpy(pData, pPtr, size);
-    else
-      memset(pData, 0, size);
+    DXASSERT_NOMSG(pContents && contentSize);
+    memcpy(pData, pContents, contentSize);
+
+    // Guarantee null-termination
+    if (bNullTerminated && newSize > contentSize) {
+      if (codePage == CP_UTF8)
+        ((char*)pData)[newSize - 1] = 0;
+      else if (codePage == DXC_CP_WIDE)
```

Comment:
What if codePage is neither CP_UTF8 or DXC_CP_WIDE? Is that impossible? Could add an else { assert(false) } in that case.